src/logger.cpp

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2016 University of Auckland
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "libcellml/logger.h"
+
+#include <vector>
+
+#include "libcellml/types.h"
+#include <libcellml/component.h>
```

Comment:
Nitpicking: swap lines 21 and 22.

There is an inconsistent use of `"` and `<`/`>` for `#include`s on lines 21 and 22, as well as throughout the code. Personally, I like to use `<`/`>` for headers that don't belong to the project on which I am working and `"` for headers that belong to it.